src/editor.rs

```diff
@@ -500,6 +500,12 @@ impl<'a, W: Write> Editor<'a, W> {
         cur_buf!(self)
     }
 
+    ///  Returns a mutable reference to the current buffer being edited.
+    /// This may be the new buffer or a buffer from history.
+    pub fn current_buffer_mut(&mut self) -> &mut Buffer {
+        cur_buf_mut!(self)
+    }
+
```

Comment:
instead of this method, I could add `start_undo_group()` and `end_undo_group()` methods to `Editor` as this is only used to call those methods on `Buffer`. Not sure if that is better.